File: MH/DAY6/10799.js
const path = process.platform === "linux" ? "/dev/stdin" : "../input.txt";
let input = require('fs').readFileSync(path).toString().trim().split('');

let stack = [];
let answer = 0;

for(var i in input){
    if(input[i] === '('){
        stack.push(input[i]);
    }else{
        if(input[i-1] === '('){
            stack.pop();
            answer += stack.length;
        }else{
            stack.pop();
            answer ++;
        }
    }
}
console.log(answer);

File: AJM/DAY11/10799.py
#10799 쇠막대기
arr = str(input())
stk = []
answer = 0

for i in range(len(arr)):
    if arr[i] == "(":
        stk.append(arr[i])
    elif arr[i] == ")":
        stk.pop()
        if arr[i-1] == "(":
            answer += len(stk)
        else:
            answer += 1

print(answer)